### PR TITLE
Support BigDecimal data type in expressions

### LINKED ISSUE #4817: Support BigDecimal data type in expressions
**Is your feature request related to a problem? Please describe.**
We use BigDecimal data type in some sources (like dynamodb) and support it in convert_entry_type processor, but it doesn't seem to work with expressions.

For example, if I run a pipeline with this processor combo:
```
    - convert_entry_type:
        key: /outer/inner
        type: "big_decimal"
    - add_entries:
        entries:
          - key: new_field
            value_expression: /outer/inner
```
with input: {"outer":{"inner": -0.31}}
I get this error:
```
org.opensearch.dataprepper.expression.ExpressionCoercionException: Unsupported type for value -0.31
        at org.opensearch.dataprepper.expression.ParseTreeCoercionService.lambda$new$0(ParseTreeCoercionService.java:33) ~[data-prepper-expression-2.9.0-SNAPSHOT.jar:?]
        at org.opensearch.dataprepper.expression.ParseTreeCoercionService.resolveJsonPointerValue(ParseTreeCoercionService.java:106) ~[data-prepper-expression-2.9.0-SNAPSHOT.jar:?]
        at org.opensearch.dataprepper.expression.ParseTreeCoercionService.coercePrimaryTerminalNode(ParseTreeCoercionService.java:69) ~[data-prepper-expression-2.9.0-SNAPSHOT.jar:?]
        at org.opensearch.dataprepper.expression.ParseTreeEvaluatorListener.visitTerminal(ParseTreeEvaluatorListener.java:69) ~[data-prepper-expression-2.9.0-SNAPSHOT.jar:?]
```

**Describe the solution you'd like**
Seems we just need to add BigDecimal here: https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/LiteralTypeConversionsConfiguration.java

**Describe alternatives you've considered (Optional)**

**Additional context**

### PR DESCRIPTION
**Pull request by kkondaka**
### Description
Support BigDecimal data type in expressions
 
### Issues Resolved
Resolves #4817 
 
### Check List
- [ X] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [ X] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ParseTreeCoercionService.java [hunk 1]
```diff
@@ -12,6 +12,7 @@
 import javax.inject.Inject;
 import javax.inject.Named;
 import java.io.Serializable;
+import java.math.BigDecimal;
```
Comment (oeyh) [inline data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ParseTreeCoercionService.java:15]: This is unused.

**Review by oeyh — COMMENTED** [review]
Thanks for working on this! 

Looks good other than an unused import.

File: data-prepper-expression/src/main/antlr/DataPrepperExpression.g4 [hunk 2]
```diff
@@ -339,6 +340,7 @@ EXPONENTLETTER
     ;
 
 INTEGER: 'integer';
+BIG_DECIMAL: 'big_decimal';
```
Comment (dlvenable) [inline data-prepper-expression/src/main/antlr/DataPrepperExpression.g4:343]: Didn't we use the term `decimal` to be consistent with OpenSearch?

**Review by dlvenable — CHANGES_REQUESTED** [review]

**Review by oeyh — APPROVED** [review]

File: data-prepper-expression/src/main/antlr/DataPrepperExpression.g4 (hunk 2 above)
Comment (dlvenable) [reply data-prepper-expression/src/main/antlr/DataPrepperExpression.g4:343]: We did use `big_decimal`, but that was already released. We should fix this by adding an alias in another PR.

**Review by dlvenable — APPROVED** [review]

**Pull request merged**